pkg/sdk/identifier_helpers.go

```diff
@@ -333,6 +334,79 @@ func (i SchemaObjectIdentifier) ArgumentsSignature() string {
 	return fmt.Sprintf("%v(%v)", i.Name(), strings.Join(arguments, ","))
 }
 
+type SchemaObjectIdentifierWithArguments struct {
+	databaseName      string
+	schemaName        string
+	name              string
+	argumentDataTypes []DataType
+}
+
+func NewSchemaObjectIdentifierWithArguments(databaseName, schemaName, name string, argumentDataTypes ...DataType) SchemaObjectIdentifierWithArguments {
+	return SchemaObjectIdentifierWithArguments{
+		databaseName:      strings.Trim(databaseName, `"`),
+		schemaName:        strings.Trim(schemaName, `"`),
+		name:              strings.Trim(name, `"`),
+		argumentDataTypes: argumentDataTypes,
+	}
+}
+
+func NewSchemaObjectIdentifierWithArgumentsInSchema(schemaId DatabaseObjectIdentifier, name string, argumentDataTypes ...DataType) SchemaObjectIdentifierWithArguments {
+	return NewSchemaObjectIdentifierWithArguments(schemaId.DatabaseName(), schemaId.Name(), name, argumentDataTypes...)
+}
+
+func NewSchemaObjectIdentifierWithArgumentsFromFullyQualifiedName(fullyQualifiedName string) (SchemaObjectIdentifierWithArguments, error) {
```

Comment:
THis one is a general question for the identifiers rework and the family of `NewXxxIdentifierFromFullyQualifiedName` -> should we have the if we prepared `ParseXxxIdentifier` helper functions?

Reply:
Not fully understand 😅, but yes there should be ParseXxxIdentifier for the new Identifier and it should be used in this function. I'll write it in the follow up pr.

Reply:
Used ParseXxxIdentifier and removed NewXxxIdentifierFromFullyQualifiedName for the new identifier